utils/utils.c

```diff
@@ -178,12 +178,45 @@ int remove_directory(char *dirname)
 	return ret;
 }
 
+#define PATH_SEPARATOR '/'
+#define DATAPATH_ROOT -2
+#define NOT_RELAVANT -1
+#define PATH_UNDER_DATAPATH 0
+
+int check_path_under_datapath(char *path, char *data_path)
+{
+        int data_path_leng = strlen(data_path);
+        int index = 0;
+
+        if (data_path_leng == 1 && data_path[0] == PATH_SEPARATOR)
+		return DATAPATH_ROOT;
+
+        while (index < data_path_leng) {
+                if (path[index] != data_path[index])
+			return NOT_RELAVANT;
+                index++;
+        }
+
+        if (path[index] == PATH_SEPARATOR || path[index] == '\00')
+                return PATH_UNDER_DATAPATH;
+
+	return NOT_RELAVANT;
+}
+
 int create_directory(char *dirname)
 {
 	int ret = -1;
 	char *oldname = NULL;
+	char *cwd_path = NULL;
+	char resolved_data_path[PATH_MAX], current_working_dir[PATH_MAX];
 
 	xasprintf(&oldname, "%s.old", dirname);
+	realpath(dirname, resolved_data_path);
+	cwd_path = getcwd(current_working_dir, PATH_MAX);
+
+	if (!check_path_under_datapath(cwd_path, resolved_data_path)) {
+		pr_err("Do not select the path which uftrace located or its parent path to save uftrace data.");
+	}
 
 	if (!access(dirname, F_OK)) {
 		if (!access(oldname, F_OK) && remove_directory(oldname) < 0) {
```

Comment:
In addition, I think it's better checking the current and old directories are actually uftrace data directory (or empty directory).  You could check it has `info` and/or `task.txt` files at least.